src/components/Search/SearchRouter/SearchRouterInput.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {useState} from 'react';
+import BaseTextInput from '@components/TextInput/BaseTextInput';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+
+type SearchRouterInputProps = {
+    isFullWidth: boolean;
+    onChange: (searchTerm: string) => void;
+    onSubmit: () => void;
+};
+
+function SearchRouterInput({isFullWidth, onChange, onSubmit}: SearchRouterInputProps) {
+    const styles = useThemeStyles();
+
+    const [value, setValue] = useState('');
+
+    const onChangeText = (text: string) => {
+        setValue(text);
+        onChange(text);
+    };
+
+    const modalWidth = isFullWidth ? styles.w100 : {width: variables.popoverWidth};
+
+    return (
+        <BaseTextInput
+            value={value}
+            onChangeText={onChangeText}
+            onSubmitEditing={onSubmit}
+            autoFocus
```

Comment:
On mWeb Safari, the autoFocus does not work well while animation is still ongoing, the caret lags behind (https://github.com/Expensify/App/issues/52382). Usually we use `useAutoFocusInput` hook instead. However since this input is not rendered in RHP (or LHP) but it's inside a modal (and the timing in modal vs RHP are diff) we had to go for another solution which is to just hide the caret while modal is still animating.